### PR TITLE
calamares: fix module functionality

### PR DESCRIPTION
**Pull request by vlinkz**
###### Motivation for this change

The calamares locale module segfaults and crashes calamares when loaded. When the keyboard module is used, no options are available. Both of these modules were broken due to hardcoded paths pointing to the wrong files. Also, the password quality checking functionality in the users module was missing.

###### Things done

Changed hardcoded paths to fix the issues in the locale and keyboard modules. Also added libpwquality dependency to enable password quality checking in the users module.

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/calamares/default.nix [hunk 1]
```diff
@@ -37,10 +37,14 @@ mkDerivation rec {
...
-        -i src/modules/locale/SetTimezoneJob.cpp
+        -i src/modules/locale/SetTimezoneJob.cpp \
+        -i src/modules/locale/Tests.cpp \
+        -i src/libcalamares/locale/TimeZone.cpp \
+        -i src/libcalamares/locale/zone-extractor.py
 
     sed -e 's,/usr/share/X11/xkb/rules/base.lst,${xkeyboard_config}/share/X11/xkb/rules/base.lst,' \
-        -i src/modules/keyboard/keyboardwidget/keyboardglobal.h
+        -i src/modules/keyboard/keyboardwidget/keyboardglobal.cpp \
+        -i src/modules/keyboard/layout-extractor.py
```
Comment (bobby285271) [inline pkgs/tools/misc/calamares/default.nix:47]: Not familiar with the codebase, but I think this file is only used to generate [`src/modules/keyboard/KeyboardData_p.cxxtr`](https://github.com/calamares/calamares/blob/calamares/src/modules/keyboard/KeyboardData_p.cxxtr) and maybe we don't need to modify it.

File: pkgs/tools/misc/calamares/default.nix [hunk 2]
```diff
@@ -37,10 +37,14 @@ mkDerivation rec {
         -i com.github.calamares.calamares.policy
 
     sed -e 's,/usr/share/zoneinfo,${tzdata}/share/zoneinfo,' \
-        -i src/modules/locale/SetTimezoneJob.cpp
+        -i src/modules/locale/SetTimezoneJob.cpp \
+        -i src/modules/locale/Tests.cpp \
+        -i src/libcalamares/locale/TimeZone.cpp \
+        -i src/libcalamares/locale/zone-extractor.py
```
Comment (bobby285271) [inline pkgs/tools/misc/calamares/default.nix:43]: And actually I think this is also just a code generator of [`src/libcalamares/locale/ZoneData_p.cxxtr`](https://github.com/calamares/calamares/blob/calamares/src/libcalamares/locale/ZoneData_p.cxxtr).

**Comment by vlinkz** [comment]
I removed changes that were unnecessary for proper execution of the program. I also added a fix for launching from the desktop.

**Comment by vlinkz** [comment]
Closed in favor of https://github.com/NixOS/nixpkgs/pull/161788

**Pull request closed without merging**